resources/assets/coffee/react/beatmapset-page/header.coffee

```diff
@@ -19,6 +19,14 @@
 el = React.createElement
 
 class BeatmapsetPage.Header extends React.Component
+  toggleFavorite: (e) ->
+    e.preventDefault()
+
+    if _.isEmpty currentUser
+      userLogin.show()
```

Comment:
there's a bug in `UserLogin#loginSuccess`. The whole action should be wrapped in a timer to delay its execution.